### PR TITLE
GHA: do not use use dci-check-change anymore

### PR DESCRIPTION
**Pull request by fredericlepied**
TestDallasWorkload: test
Test-Hints: force-check

### CODE REVIEW

File: hack/maybe-dci-check-change
Comment (ramperher) [inline hack/maybe-dci-check-change:1]: Does it make sense to keep calling this (...)-dci-check-change if we're removing dci-check-change support?

File: hack/dci-merge.sh [hunk 1]
```diff
@@ -54,100 +45,43 @@ send_status() {
...
-
-            # process Test-App-Args-Hints
-            if [[ app-args =~ $SUPPORTED_HINTS ]]; then
-                APP_ARGS=$(sed -n -e "s/^\s*Test-App-Args-Hints:\s*//pi" <<< "$DESC")
-            fi
-
-            # stop at the first valid Test-Hints: string
-            break
+        # extract TestBos2Sno commands
+        if grep -qE "^\s*TestBos2Sno:\s*" <<< "$DESC"; then
```
Comment (ramperher) [inline hack/dci-merge.sh:77]: I'm wondering where are these TestBos2Sno/TestBos2Baremetal vars defined in the docs, just to have clear how to use them. Are they similar to TestWorkload or TestDallas?

File: docs/dev.md [hunk 2]
```diff
@@ -55,14 +55,33 @@ Edit `MAJOR` when there are breaking changes in the collection. Edit `MINOR` whe
...
-- `check workload preflight-green`
+  * `TestDallas`: baremetal clusters in the Dallas lab.
+  * `TestDallasWorkload`: workload on a pre-installed baremetal cluster in Dallas.
+  * `TestBos2`: virtual setup in the BOS2 lab.
+  * `TestBos2Sno`: virtual SNO setup in the BOS2 lab.
+  * `TestBos2SnoBaremetal`: baremetal SNO node in the BOS2 lab.
+
+The following `Test-Hints` can be specified if needed in the description of the PR:
+
+  * `Test-Hint: no-check` when there is no test is needed in BOS2. For a doc only change this is detected automatocally.
```
Comment (tonyskapunk) [inline docs/dev.md:72]:
```suggestion
  * `Test-Hint: no-check` when there is no test is needed in BOS2. For a doc only change this is detected automatically.
```

Comment (fredericlepied) [reply docs/dev.md]: fixed

File: docs/dev.md [hunk 3]
```diff
@@ -55,14 +55,33 @@ Edit `MAJOR` when there are breaking changes in the collection. Edit `MINOR` whe
...
-- DCI Dallas pipeline: run DCI jobs on a baremetal cluster in the Telco Partner CI Dallas lab. It is triggered only when changes are modifying files in the `role/cnf-cert` or `roles/preflight` directories.
+- DCI BOS2 pipeline: [`.github/workflows/dci.yml`](../.github/workflows/dci.yml) that runs a DCI job to test the collection in a virtual environment at the BOS2 Telco Partner CI lab. It is triggered only when a change is modifying files in the `/roles/` directory.
+- DCI Dallas pipeline: run DCI jobs on a baremetal cluster in the Telco Partner CI Dallas lab. It is triggered automatically only when changes are modifying files in the `role/cnf-cert` or `roles/preflight` directories.
 - RPM build pipeline (local/check). This is managed by the Zuul CI on https://softwarefactory-project.io/. It builds rpm for el8 and el9.
 
-If you are part of the project, you can also use comments in the PR to trigger the baremetal pipelines manually in the Telco Partner CI Dallas lab:
+To specify which DCI lab to use and which resources to use, you can use the following strings in the descrption of the PR:
```
Comment (ramperher) [inline docs/dev.md:62]: Maybe it's worth to include here that these strings are built based on dci-auto-launch docs from https://docs.distributed-ci.io/dci-pipeline/#dci-auto-launch

File: hack/maybe-dci-check-change
Comment (ramperher) [reply hack/maybe-dci-check-change:1]: this still applies

Comment (fredericlepied) [reply hack/maybe-dci-check-change:1]: It is difficult to rename jobs so that's why I didn't want to change the name.

**Comment by fredericlepied** [comment]
retest

File: docs/dev.md (hunk 3 above)
Comment (fredericlepied) [reply docs/dev.md:62]: I don't think it could help contributors to the collection.

File: hack/maybe-dci-check-change [hunk 4]
```diff
@@ -35,18 +35,33 @@ pr=$(sed -e 's@.*/pull/'@@ <<< $URL)
...
+# check if there is a Test-Hints: no-check string in the PR description
+if grep -iEq '^\s*Test-Hints?:\s*no-check' <<< $body; then
+    no_check=1
+else
+    no_check=
+fi
+
 # Check if there is a force-check or a code change
-if [ -n "$force_check" ] || curl -s -L "${HEADERS[@]}" "https://api.github.com/repos/$proj/pulls/$pr/files"|jq -r .[].filename|grep -Ev '^tests/|\.md$'|grep -E 'roles/|plugins/|galaxy.yml'; then
+if [ -z "$no_check" ] && ( [ -n "$force_check" ] || curl -s -L "${HEADERS[@]}" "https://api.github.com/repos/$proj/pulls/$pr/files"|jq -r .[].filename|grep -Ev '^tests/|\.md$'|grep -E 'roles/|plugins/|galaxy.yml' ); then
```
Comment (tonyskapunk) [inline hack/maybe-dci-check-change:55]: nit: muti-line

```suggestion
if [ -z "$no_check" ] &&
   ( [ -n "$force_check" ] ||
     curl -s -L "${HEADERS[@]}" "https://api.github.com/repos/$proj/pulls/$pr/files" |
     jq -r .[].filename |
     grep -Ev '^tests/|\.md$' |
     grep -E 'roles/|plugins/|galaxy.yml' 
   ); then
```

**Review by tonyskapunk — APPROVED** [review]
LGTM

**Review by tonyskapunk — APPROVED** [review]

**Pull request merged**